src/databricks/labs/remorph/reconcile/query_builder/threshold_query.py

```diff
@@ -175,3 +175,13 @@ def _build_threshold_percentage_case(cls, base: exp.Expression, threshold: Thres
             true=exp.Identifier(this="Warning", quoted=True),
         )
         return exp.Case(ifs=[eq_if, between_if], default=exp.Identifier(this="Failed", quoted=True))
+
+    def build_threshold_query(self) -> str:
+        keys: list[str] = sorted(self.partition_column.union(self.join_columns).union(self.threshold_columns))
+        select_clause = [
+            build_column(this=col, alias=self.table_conf.get_tgt_to_src_col_mapping(col, self.layer)) for col in keys
+        ]
+        select_clause = self.add_transformations(select_clause, self.source)
+        query = select(*select_clause).from_(":tbl").where(self.filter)
```

Comment:
why not the actual table name here?

Reply:
for table name we need source catalog and schema info also .. but the constructor for QueryBuilder only gets table_conf: Table (this dosent have catalog info) So .. we pass it like this
and in the integration layer .. it will be replaced by proper table name .. thats how sampling query is alos built

Reply:
constructor needs to be changed